core/api/service/api_service.cpp

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "api/service/api_service.hpp"
+#include "api/jrpc/jrpc_processor.hpp"
+
+namespace kagome::api {
+
+  ApiService::ApiService(std::shared_ptr<Listener> listener,
+                         std::shared_ptr<JRPCProcessor> processor)
+      : listener_(std::move(listener)), processor_(std::move(processor)) {
+    listener_->onError().connect([this](outcome::result<void> err) { stop(); });
+  }
+
+  void ApiService::start() {
+    // handle new session
+    listener_->start([self =
```

Comment:
The double start is now possible. Won't it break something?

Reply:
fixed